### DISCUSSION TITLE
package-lock.json handling in workspaces

### DISCUSSION THREAD

**Discussion by jasonwilliams** [npm/feedback#625] [CLI] [answered]
When using workspaces in yarn if you update a dep in a sub package, only the top package-lock gets updated. However in npm it adds a new package-lock.json in the sub package even though im workiing in a workspace. is there any way to prevent this?

It feels like a bug as it only seems to happen now and again. Is there a way for npm to detect its in a workspace and only update the top level package?

**Comment by MylesBorins** [comment]
@jasonwilliams I think the appropriate next step would be to open an issue in the npm CLI repo if you think this is a bug. If you think it is intended behavior we should change an issue in the npm RFC repo would be more appropriate.

**Reply by jasonwilliams** [reply]
Raised https://github.com/npm/cli/issues/4226

**Reply by leobalter** [reply, marked as answer]
The bug seems fixed by npm/cli#4226. Thanks!
